### PR TITLE
Enable ReleaseChannel in AutoUpdate

### PR DESCRIPTION
**Pull request by DaveTryon**
#### Describe the change
AutoUpdate needs to be responsive to ReleaseChannel settings (it currently only supports Production)

#### PR checklist

- [ ] Run through of all [test scenarios](https://github.com/Microsoft/accessibility-insights-windows/blob/master/docs/Scenarios.md) completed?
- [ ] Does this address an existing issue? If yes, Issue# - 
- [ ] Includes UI changes?
  - [ ] Run the production version of Accessibility Insights for Windows against a version with changes.
  - [ ] Attach any screenshots / GIF's that are applicable.

> Note: After the PR has been created, certain checks will be kicked off. All of these checks must pass before a merge.

### CODE REVIEW

File: src/AccessibilityInsights.Extensions.GitHubAutoUpdate/ProductionChannelInfoProvider.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ public ProductionChannelInfoProvider(IGitHubWrapper gitHubWrapper, IExceptionRep
         /// </summary>
         public bool TryGetChannelInfo(ReleaseChannel releaseChannel, out ChannelInfo channelInfo)
         {
-            return ChannelInfoUtilities.TryGetChannelInfo(releaseChannel, out channelInfo, _gitHubWrapper, null, _exceptionReporter);
```
Comment (DaveTryon) [inline src/AccessibilityInsights.Extensions.GitHubAutoUpdate/ProductionChannelInfoProvider.cs]: >null [](start = 107, length = 4)

Passing null here was a bug. We want to use the default parameter, hence the named parameter syntax

File: src/AccessibilityInsights/MainWindowHelpers/Initialize.cs [hunk 2]
```diff
@@ -114,6 +115,9 @@ private static void PopulateConfigurations()
             // when AccessibilityInsights starts up. 
             ConfigurationManager.GetDefaultInstance().AppConfig.IsUnderElementScope = true;
 
+            // Configure the correct ReleaseChannel for autoupdate
+            Container.SetAutoUpdateReleaseChannel(ConfigurationManager.GetDefaultInstance().AppConfig.ReleaseChannel.ToString());
+
             // enable/disable telemetry
             if (ConfigurationManager.GetDefaultInstance().AppConfig.EnableTelemetry)
                 TelemetryController.EnableTelemetry();
```
Comment (DaveTryon) [inline src/AccessibilityInsights/MainWindowHelpers/Initialize.cs]: >TelemetryController [](start = 16, length = 19)

This is where the Container currently gets initialized, so we need to set the ReleaseChannel before reaching this point

File: src/AccessibilityInsights.Extensions/Container.cs [hunk 3]
```diff
@@ -145,6 +145,16 @@ public static Container GetDefaultInstance()
 
             return _defaultInstance;
         }
+
+        /// <summary>
+        /// Set the ReleaseChannel for update
+        /// </summary>
+        /// <param name="releaseChannel">The value to use in the AutoUpdate code</param>
+        public static void SetAutoUpdateReleaseChannel(string releaseChannel)
```
Comment (DaveTryon) [inline src/AccessibilityInsights.Extensions/Container.cs]: >string [](start = 55, length = 6)

This uses a string to keep Extensions definitions decoupled from the SetupLibrary extension

File: src/AccessibilityInsights.Extensions.GitHubAutoUpdate/AutoUpdate.cs [hunk 4]
```diff
@@ -170,26 +170,37 @@ private UpdateResult Update()
...
         {
-            _strongReleaseChannel = releaseChannel;
+            _strongReleaseChannel = releaseChannelProvider();
             _installedVersionProvider = installedVersionProvider;
             _channelInfoProvider = channelInfoProvider;
             ReleaseChannel = _strongReleaseChannel.ToString();
             _initTask = Task.Run(() => InitializeWithTimer());
         }
 
+        private static ReleaseChannel ProductionReleaseChannelProvider()
```
Comment (karanbirsingh) [inline src/AccessibilityInsights.Extensions.GitHubAutoUpdate/AutoUpdate.cs]: can we consider renaming so it doesn't have Production in the name?

Comment (DaveTryon) [reply src/AccessibilityInsights.Extensions.GitHubAutoUpdate/AutoUpdate.cs]: Renamed to ConfiguredReleaseChannelProvider

---
In reply to: [273264454](https://github.com/Microsoft/accessibility-insights-windows/pull/277#discussion_r273264454) [](ancestors = 273264454)

**Pull request merged**